src/EStimLibrary/Extensions/SpatialModel/StringHierarchy/StringHierarchyLocationFactory.cs

```diff
@@ -28,36 +51,67 @@ public StringHierarchyLocationFactory(
 
         this.ParamLimits = new() {
             {"fullSpec",
-                new DynamicDataLimits<string>(this.LocationSpecCheckFunction,
+                new DynamicDataLimits<string>(this._LocationSpecCheckFunction,
                 this.HelpMsg) } };
-
     }
 
-    private bool LocationSpecCheckFunction(string fullSpec)
+    /// <summary>
+    /// The validation function used by the factory create method to ensure
+    /// the supplied location spec - region spec and modifiers - is valid 
+    /// within the provided base region.
+    /// </summary>
+    /// <param name="fullSpec">The full location spec to validate.</param>
+    /// <returns>True if valid location spec, false if not.</returns>
+    private bool _LocationSpecCheckFunction(string fullSpec)
     {
         var parts = fullSpec.Split(
             StringHierarchySpec.REGIONS_MODIFIERS_DELIMITER);
 
-        return parts.Length > 0 &&
+        return parts.Length > 0 && parts.Length < 3 &&
             // First try to navigate this model to the specified region.
             this._baseRegion.TryGetSubregion(parts[0], out var subregion) &&
             // Then - if any given - check if the modifiers valid in the model.
-            (parts.Length > 1) ?
-            subregion.IsValidModifierSpec(parts[1]) : true;
+            ((parts.Length > 1) ?
+                subregion.IsValidModifierSpec(parts[1]) : true);
     }
 
+    /// <summary>
+    /// Create a StringHierarchyLocation object from the provided parameter
+    /// values.
+    /// </summary>
+    /// <param name="paramValues">The parameters to pass into the
+    /// StringHierarchyLocation constructor. This must adhere to the 
+    /// ParamLimits of this factory for successful creation, i.e., include a
+    /// single element keyed "fullSpec" which is a full string location spec 
+    /// valid within this factory's base region.</param>
+    /// <param name="product">An output parameter: the StringHierarchyLocation
+    /// if created, else null.</param>
+    /// <param name="skipValueValidation">An optional parameter: perform only
+    /// basic input validation and skip any deeper parameter value validation,
+    /// e.g., if the data has already been validated. Default: false.</param>
+    /// <returns>True if a StringHierarchyLocation could be created, False if 
+    /// not.</returns>
     public bool TryCreate(Dictionary<string, object> paramValues,
-        out ILocation product, bool skipValueValidation = false)
+        out ILocation? product, bool skipValueValidation = false)
```

Comment:
Added ? to indicate this output param could be null, an added ! in the function body to suppress null warnings (since the function value is passed to can handle null values, or value will have been checked to be non-null by the time it has used). You will need to similarly use ? and ! in the test classes to declare variables that could be null (i.e., your variables for output params) and suppress null value warnings (in my VSCode, all the places warnings appear show as yellow-underlined which could help identify them). I started changing some places in the SHLocationFactoryTests.cs class, but only barely --> will need to do the rest